src/Common/Http/Client.php

```diff
@@ -2,47 +2,78 @@
 
 namespace Omnipay\Common\Http;
 
-use function GuzzleHttp\Psr7\str;
-use Http\Client\HttpClient;
-use Http\Discovery\HttpClientDiscovery;
-use Http\Discovery\MessageFactoryDiscovery;
-use Http\Message\RequestFactory;
+use Http\Discovery\Psr17FactoryDiscovery;
+use Http\Discovery\Psr18ClientDiscovery;
 use Omnipay\Common\Http\Exception\NetworkException;
 use Omnipay\Common\Http\Exception\RequestException;
 use Psr\Http\Message\RequestInterface;
 use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\StreamFactoryInterface;
 use Psr\Http\Message\StreamInterface;
-use Psr\Http\Message\UriInterface;
+use Psr\Http\Client\ClientInterface as HttpClientInterface;
+use Psr\Http\Message\RequestFactoryInterface;
 
 class Client implements ClientInterface
 {
     /**
      * The Http Client which implements `public function sendRequest(RequestInterface $request)`
-     * Note: Will be changed to PSR-18 when released
      *
-     * @var HttpClient
+     * @var HttpClientInterface
      */
     private $httpClient;
 
     /**
-     * @var RequestFactory
+     * @var RequestFactoryInterface
      */
     private $requestFactory;
 
-    public function __construct($httpClient = null, RequestFactory $requestFactory = null)
+    /**
+     * @var StreamFactoryInterface
+     */
+    private $streamFactory;
+
+    public function __construct($httpClient = null, $requestFactory = null, $streamFactory = null)
```

Comment:
Should these all be type-hinted? I'm not sure if a later PHP breaks this by requiring the typehint to be optional e,g. `?StreamFactoryInterface` and that's the reason for leaving optional parameters unhinted? Just discovering this myself.

```suggestion
    public function __construct(HttpClientInterface $httpClient = null, RequestFactoryInterface$ requestFactory = null, StreamFactoryInterface $streamFactory = null)
```

Reply:
The reason they are not typehinted, is that current versions use HttpPlug instead of PSR Http Client, so that would be a breaking change.